Core/src/org/sleuthkit/autopsy/datamodel/OsAccounts.java

```diff
@@ -208,38 +209,50 @@ public void propertyChange(PropertyChangeEvent evt) {
                         && evt.getNewValue() instanceof AsynchOsAcctData
                         && ((AsynchOsAcctData) evt.getNewValue()).getOsAccountId() == account.getId()) {
 
+                    List<NodeProperty<?>> propertiesToUpdate = new ArrayList<>();
+
                     AsynchOsAcctData osAcctData = (AsynchOsAcctData) evt.getNewValue();
 
                     List<String> realmNames = osAcctData.getOsAcctRealm().getRealmNames();
                     if (!realmNames.isEmpty()) {
                         String realmNamesStr = realmNames.stream()
                                 .map(String::trim)
                                 .distinct()
-                                .sorted((a,b) -> a.compareToIgnoreCase(b))
+                                .sorted((a, b) -> a.compareToIgnoreCase(b))
                                 .collect(Collectors.joining(", "));
 
-                        updateSheet(new NodeProperty<>(
+                        propertiesToUpdate.add(new NodeProperty<>(
                                 Bundle.OsAccounts_accountRealmNameProperty_name(),
                                 Bundle.OsAccounts_accountRealmNameProperty_displayName(),
                                 Bundle.OsAccounts_accountRealmNameProperty_desc(),
                                 realmNamesStr));
                     }
 
+                    String scopeName = osAcctData.getOsAcctRealm().getScope().getName();
+                    if (StringUtils.isNotBlank(scopeName)) {
+                        propertiesToUpdate.add(new NodeProperty<>(
+                                Bundle.OsAccounts_accountScopeNameProperty_name(),
+                                Bundle.OsAccounts_accountScopeNameProperty_displayName(),
+                                Bundle.OsAccounts_accountScopeNameProperty_desc(),
+                                scopeName));
+                    }
+
                     List<Host> hosts = osAcctData.getHosts();
                     if (!hosts.isEmpty()) {
                         String hostsString = hosts.stream()
                                 .map(h -> h.getName().trim())
                                 .distinct()
-                                .sorted((a,b) -> a.compareToIgnoreCase(b))
+                                .sorted((a, b) -> a.compareToIgnoreCase(b))
                                 .collect(Collectors.joining(", "));
 
-                        updateSheet(new NodeProperty<>(
+                        propertiesToUpdate.add(new NodeProperty<>(
                                 Bundle.OsAccounts_accountHostNameProperty_name(),
                                 Bundle.OsAccounts_accountHostNameProperty_displayName(),
                                 Bundle.OsAccounts_accountHostNameProperty_desc(),
                                 hostsString));
                     }
 
+                    updateSheet(propertiesToUpdate.toArray(new NodeProperty<?>[propertiesToUpdate.size()]));
```

Comment:
Do we need to put the updateSheet() calls into the EDT or does NetBeans do that when we call setSheet(createSheet())?

Reply:
I'm not seeing anything pushing the sheet call to the EDT so I wrapped the call in a `SwingUtilities.invokeLater` call.